src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/OtlpRetry.cs

```diff
@@ -0,0 +1,248 @@
+/// exponential backoff strategy: initial backoff, max backoff, backoff
+/// multiplier, and max retry attempts. The OTLP specification does not declare
+/// a similar set of parameters, so this implementation uses fixed settings.
+/// Furthermore, since the OTLP spec does not specify a max number of attempts,
+/// this implementation will retry until the deadline is reached.
+///
+/// The throttling mechanism for OTLP differs from the throttling mechanism
+/// described in the gRPC retry specification. See:
+/// https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#otlpgrpc-throttling.
+/// </summary>
+internal static class OtlpRetry
+{
+    public const string GrpcStatusDetailsHeader = "grpc-status-details-bin";
+    public const int InitialBackoffMilliseconds = 1000;
+    private const int MaxBackoffMilliseconds = 5000;
+    private const double BackoffMultiplier = 1.5;
+
+#if !NET6_0_OR_GREATER
+    private static readonly Random Random = new Random();
+#endif
+
+    public static bool TryGetHttpRetryResult(HttpStatusCode statusCode, DateTime? deadline, HttpResponseHeaders responseHeaders, int retryDelayMilliseconds, out RetryResult retryResult)
+    {
+        return OtlpRetry.TryGetRetryResult(statusCode, IsHttpStatusCodeRetryable, deadline, responseHeaders, TryGetHttpRetryDelay, retryDelayMilliseconds, out retryResult);
+    }
+
+    public static bool TryGetGrpcRetryResult(StatusCode statusCode, DateTime? deadline, Metadata trailers, int retryDelayMilliseconds, out RetryResult retryResult)
+    {
+        return OtlpRetry.TryGetRetryResult(statusCode, IsGrpcStatusCodeRetryable, deadline, trailers, TryGetGrpcRetryDelay, retryDelayMilliseconds, out retryResult);
+    }
+
+    private static bool TryGetRetryResult<TStatusCode, TCarrier>(TStatusCode statusCode, Func<TStatusCode, bool, bool> isRetryable, DateTime? deadline, TCarrier carrier, Func<TStatusCode, TCarrier, TimeSpan?> throttleGetter, int nextRetryDelayMilliseconds, out RetryResult retryResult)
+    {
+        retryResult = default;
+
+        // TODO: Consider introducing a fixed max number of retries (e.g. max 5 retries).
+        // The spec does not specify a max number of retries, but it may be bad to not cap the number of attempts.
+        // Without a max number of retry attempts, retries would continue until the deadline.
+        // Maybe this is ok? However, it may lead to an unexpected behavioral change. For example:
+        //    1) When using a batch processor, a longer delay due to repeated
+        //       retries up to the deadline may lead to a higher chance that the queue will be exhausted.
+        //    2) When using the simple processor, a longer delay due to repeated
+        //       retries up to the deadline will lead to a prolonged blocking call.
+        // if (attemptCount >= MaxAttempts)
+        // {
+        //     return false
+        // }
```

Comment:
I'm inclined to set a maximum cap of 5 retries, but I'd like to get other people's thoughts.

Reply:
5 sounds good. 
`1)`, users can adjust the queue length if they frequently see dropped items. for `2)` there is no mitigation given the retries are not configurable, correct?

Reply:
> 5 sounds good.

Yes, I agree. I plan to leave this as a follow up for now. Put it in place when I actually wire this up with the exporter.